g-local_vce_detailed.md

```diff
@@ -318,3 +318,32 @@ The following flags may also be used with the `docker run` command:
 | Automatically remove container when it exits  | `--rm`                                                   |
 | Run container in background      | `-d/--detach`                                            |
 | Set environment variable inside  container     | `-e/--env ENV_VARIABLE="ENV VALUE"`                      |
+## Updating the Docker application
+It is good practice to keep your version of Docker Desktop up to date. If there are updates to Docker Desktop, the *Update to latest* button will appear to the left of the Docker Desktop search toolbar, {numref}`docker_desktop_update`.
+
+```{figure} md_assets/media/docker_update.png
+:name: docker_desktop_update
+:width: 5.44791666666667in
+:height: 2.041666666667in
+
+Checking for updates in Docker Desktop
+
+```
+It is possible to update to the latest version by doing the following.
+- Click on the *Update to latest* button. This will take you to the *Softwares updates* page.
+- Click on *Download update* and then *Update and restart*.
+
+Note: we strongly recommend that you **do not update the software within the VCE unless told to do so by the module team**. Software updates and upgrades occasionally introduce changes that result in undesirable software behaviour, sometimes known as ‘breaking changes’, that cannot always be predicted in advance.
```

Comment:
This is a bit confusing; you have been talking about the Docker application but no you switch to the s/w inside the container on the hosted VCE.

The same should apply to not updating the s/w inside a container unless told to do so (because this can break the environment). Ideally, to update s/w inside the image, we would ship a new image (this is what happens on the hosted VCE) and then tell students to update their image and create a new container.

In passing, I note the mockery of saying:

- you should update Docker if it says an update is available
- you should update the module docker image if we tell you
- you should ignore all warnings about out of date software inside the docker container...

Reply:
You're right that it does sound inconsistent (and is inconsistent!). I was trying to get the balance between complying with the guidance from Information Security and what is 'safe' from a module perspective. Maybe the answer is to tone down what we say about updating Docker. Say, to merely 'if they want to'?

Reply:
so s/thing like: upgrading Docker often improves performance and introduces new features, as well as fixing issues, but is often not essential. If you notice that your Docker application is not working correctly, and update may fix it. (You can always check the [release notes](https://docs.docker.com/release-notes/) for *Docker Desktop* and *Dokcer Engine* to see what changes are associated with any given release.)